src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3061,26 +3061,30 @@ ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> rec
                 int addedCount = 0;
                 foreach (ParameterSymbol param in recordParameters)
                 {
-                    var property = new SynthesizedRecordPropertySymbol(this, param, diagnostics);
-                    _ = memberSignatures.TryGetValue(property, out var existingMember);
-                    existingMember ??= getInheritedMember(property, this);
+                    bool isInherited = false;
+                    var property = new SynthesizedRecordPropertySymbol(this, param, isOverride: false, diagnostics);
+                    if (!memberSignatures.TryGetValue(property, out var existingMember))
+                    {
+                        existingMember = getInheritedMember(property, this);
+                        isInherited = true;
+                    }
                     if (existingMember is null)
                     {
-                        existingOrAddedMembers.Add(property);
-                        members.Add(property);
-                        members.Add(property.GetMethod);
-                        members.Add(property.SetMethod);
-                        members.Add(property.BackingField);
-
-                        builder.InstanceInitializersForRecordDeclarationWithParameters.Insert(addedCount, new FieldOrPropertyInitializer.Builder(property.BackingField, paramList.Parameters[param.Ordinal]));
-                        addedCount++;
+                        addProperty(property);
                     }
                     else if (existingMember is PropertySymbol { IsStatic: false, GetMethod: { } } prop
                         && prop.TypeWithAnnotations.Equals(param.TypeWithAnnotations, TypeCompareKind.AllIgnoreOptions))
                     {
                         // There already exists a member corresponding to the candidate synthesized property.
-                        // The deconstructor is specified to simply assign from this property to the corresponding out parameter.
-                        existingOrAddedMembers.Add(prop);
+                        if (isInherited && existingMember.IsAbstract)
+                        {
+                            addProperty(new SynthesizedRecordPropertySymbol(this, param, isOverride: true, diagnostics));
```

Comment:
>addProperty(new SynthesizedRecordPropertySymbol(this, param, isOverride: true, diagnostics)); [](start = 28, length = 93)

It doesn't look like we are properly inheriting custom modifiers for the property and the accessors. We should do what is done for regular properties and their accessors.

Reply:
Added a test for now, and will wait for [https://github.com/dotnet/roslyn/pull/44883](https://github.com/dotnet/roslyn/pull/44883).

---
In reply to: [443270124](https://github.com/dotnet/roslyn/pull/45336#discussion_r443270124) [](ancestors = 443270124)